sdk/tables/aztable/table_client_test.go

```diff
@@ -46,435 +44,271 @@ func (s *tableClientLiveTests) TestServiceErrors() {
+	require.NoError(err)
+
+	var unMarshalledPreMerge map[string]interface{}
+	err = json.Unmarshal(preMerge.Value, &unMarshalledPreMerge)
+	require.NoError(err)
 
-	_, updateErr := client.UpdateEntity(ctx, mergeProperty, nil, Merge)
-	assert.Nil(updateErr)
+	var mapEntity map[string]interface{}
+	err = json.Unmarshal(*marshalled, &mapEntity)
+	require.NoError(err)
+	mapEntity["MergeProperty"] = "foo"
 
-	pager = client.Query(&QueryOptions{Filter: &filter})
+	reMarshalled, err := json.Marshal(mapEntity)
+	require.NoError(err)
+
+	_, updateErr := client.UpdateEntity(ctx, reMarshalled, nil, Merge)
+	require.Nil(updateErr)
+
+	var qResp TableEntityQueryByteResponseResponse
+	pager := client.Query(queryOptions)
 	for pager.NextPage(ctx) {
 		qResp = pager.PageResponse()
 	}
 	postMerge := qResp.TableEntityQueryResponse.Value[0]
+	var unmarshalledPostMerge map[string]interface{}
+	err = json.Unmarshal(postMerge, &unmarshalledPostMerge)
+	require.NoError(err)
+
+	require.Equal(unmarshalledPostMerge["PartitionKey"], unMarshalledPreMerge["PartitionKey"])
+	require.Equal(unmarshalledPostMerge["MergeProperty"], "foo")
 
-	// The merged entity has all its properties + the merged property
-	assert.Equalf(len(preMerge)+1, len(postMerge), "postMerge should have one more property than preMerge")
-	assert.Equalf(postMerge[mergeProp], val, "%s property should equal %s", mergeProp, val)
+	_, ok := unMarshalledPreMerge["MergeProperty"]
+	require.False(ok)
 }
 
 func (s *tableClientLiveTests) TestUpsertEntity() {
-	assert := assert.New(s.T())
 	require := require.New(s.T())
 	client, delete := s.init(true)
 	defer delete()
 
-	entitiesToCreate := createSimpleEntities(1, "partition")
+	// 1. Create Basic Entity
+	entityToCreate := createSimpleEntity(1, "partition")
+	marshalled := marshalBasicEntity(entityToCreate, require)
 
-	_, err := client.UpsertEntity(ctx, (*entitiesToCreate)[0], Replace)
-	require.Nil(err)
+	_, err := client.UpsertEntity(ctx, *marshalled, Replace)
+	require.NoError(err)
 
-	var qResp TableEntityQueryResponseResponse
 	filter := "RowKey eq '1'"
-	pager := client.Query(&QueryOptions{Filter: &filter})
-	for pager.NextPage(ctx) {
-		qResp = pager.PageResponse()
-	}
-	preMerge := qResp.TableEntityQueryResponse.Value[0]
-
-	mergeProp := "MergeProperty"
-	val := "foo"
-	var mergeProperty = map[string]interface{}{
-		partitionKey: (*entitiesToCreate)[0][partitionKey],
-		rowKey:       (*entitiesToCreate)[0][rowKey],
-		mergeProp:    val,
-	}
+	query := &QueryOptions{Filter: &filter}
 
-	_, updateErr := client.UpsertEntity(ctx, mergeProperty, Replace)
-	require.Nil(updateErr)
+	// 2. Query for basic Entity
+	preMerge, err := client.GetEntity(ctx, entityToCreate.PartitionKey, entityToCreate.RowKey, nil)
+	require.NoError(err)
+
+	var unMarshalledPreMerge map[string]interface{}
+	err = json.Unmarshal(preMerge.Value, &unMarshalledPreMerge)
+	require.NoError(err)
```

Comment:
Is how developers would have to handle entity responses, or is this just for testing purposes?

Reply:
There are MANY ways customers can handle entity responses. They unmarhsal to a map[string]interface or to a map[string]json.RawMessage, or to an EdmEntity (our own type) or to a structure that the customer defines. We enable all this flexibility by returning the raw JSON and then the customer decides how they want to deal with it.